fred2/management.cpp

```diff
@@ -146,15 +146,16 @@ int create_ship(matrix *orient, vec3d *pos, int ship_type);
 int query_ship_name_duplicate(int ship);
 char *reg_read_string( char *section, char *name, char *default_value );
 
+// Note that the parameter here is max *length*, not max buffer size.  Leave room for the null-terminator!
 void string_copy(char *dest, const CString &src, size_t max_len, int modify)
 {
 	if (modify)
 		if (strcmp(src, dest))
 			set_modified();
 
 	auto len = strlen(src);
-	if (len >= max_len)
-		len = max_len - 1;
+	if (len > max_len)
+		len = max_len;
```

Comment:
Should we log the fact that we're clamping `len`?

Reply:
I don't think so.  Truncation isn't really the sort of thing you need to log in these cases.